File: array/leetcode_189.kt
package problem_solving

class Solution189 {
    fun rotate(nums: IntArray, k: Int): Unit {
        val tmp = IntArray(nums.size)
        val n = nums.size
        for(i in nums.indices) {

            tmp[(i+k)%n] = nums[i]
        }

        for(i in nums.indices) {
            nums[i] = tmp[i]
        }


    }
}

File: array/leetcode_189.py
# [Try1: Brute Force, 시간초과]
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        for _ in range(k):
            num = nums[-1]
            tmp = nums[0]

            for i in range(1, len(nums)):
                nums[i], tmp = tmp, nums[i]

            nums[0] = num

# [Try2: 통과]
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """

        n = len(nums)
        tmp = [0] * n

        for i in range(n):
            tmp[(i + k) % n] = nums[i]

        nums[:] = tmp
